build.gradle

```diff
@@ -120,51 +120,36 @@ jmh {
 }
 
 test {
-    testLogging  {
-        // showing skipped occasionally should prevent CI timeout due to lack of standard output
-        events=["skipped", "failed"] // "started", "passed"
-        // showStandardStreams = true
-        exceptionFormat="full"
-
-        debug.events = ["skipped", "failed"]
-        debug.exceptionFormat="full"
-
-        info.events = ["failed", "skipped"]
-        info.exceptionFormat="full"
-        
-        warn.events = ["failed", "skipped"]
-        warn.exceptionFormat="full"
-    }
-
     maxHeapSize = "1200m"
-
-    if (System.getenv("CI") == null) {
-        maxParallelForks = Runtime.runtime.availableProcessors().intdiv(2) ?: 1
-    }
 }
 
-task testng(type: Test) { 
+task testNG(type: Test) {
      useTestNG()
-     testLogging  {
-        events=["skipped", "failed"]
-        exceptionFormat="full"
+}
+
+check.dependsOn testNG
+
+tasks.withType(Test) {
+    testLogging  {
+        events = ["skipped", "failed"]
+        exceptionFormat = "full"
 
         debug.events = ["skipped", "failed"]
-        debug.exceptionFormat="full"
+        debug.exceptionFormat = "full"
 
         info.events = ["failed", "skipped"]
-        info.exceptionFormat="full"
-        
+        info.exceptionFormat = "full"
+
         warn.events = ["failed", "skipped"]
-        warn.exceptionFormat="full"
-     }
-} 
+        warn.exceptionFormat = "full"
+    }
 
-check.dependsOn testng 
+    maxParallelForks = Runtime.runtime.availableProcessors().intdiv(2) ?: 1
```

Comment:
This was supposed to run parallel only in non-CI environments.

Reply:
I understand that was the intent, but it seems that it has not for quite some time. I wasn't able to find "CI" set anywhere. Should testing on Github be single threaded?

Reply:
https://docs.github.com/en/actions/reference/environment-variables#default-environment-variables
<html>
<body>
<!--StartFragment-->

Environment variable | Description
-- | --
CI | Always set to true.

<!--EndFragment-->
</body>
</html>

Reply:
Ah, thanks. I wasn't aware of that. I'll submit an update.